### PR TITLE
ResourceZone

### LINKED ISSUE #260: Create a ResourceZone for each Server Profile Template
Include list of ResourceBlocks of Server Hardware instances related to that Server Profile Template and link to the ResourceBlock of that Server Profile Template.

To get SH for a SPT (example):
```
curl -XGET -H "auth:${AUTH}" -H "X-Api-Version:300" -H "Content-Type:application/json" -k https://ai-cicvc3-004.vse.rdlabs.hpecorp.net/rest/server-profiles/available-targets?serverHardwareTypeUri=/rest/server-hardware-types/61B7D4BA-3B37-4BBB-9D97-3D2607E9BBA0&enclosureGroupUri=/rest/enclosure-groups/bc41f38d-e8ce-4241-acd1-00b2d8c5d0fa&_=1527189958171 | jq '.' 
```

### PR DESCRIPTION
**Pull request by ricardogpsf**
This PR Creates a Zone for a Server Profile Template requested from Oneview and populates the Links of Zone with available targets of the Server Profile Template.

Associated github issues are #260 and #237 

Closes #260 #237

### CODE REVIEW

**Comment by ricardogpsf** [comment]
@karolcatunda marking you on this review.

File: oneview_redfish_toolkit/api/zone.py [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+        uuid = original_dict[uri_key].split("/")[-1]
+        dict_item = dict()
+        dict_item["@odata.id"] = ResourceBlockCollection.BASE_URI + "/" + uuid
+        self.redfish["Links"]["ResourceBlocks"].append(dict_item)
+
+    def fill_capabilities_collection(self):
+        capability = {
+            "CapabilitiesObject": {
+                "@odata.id": "/redfish/v1/Systems/Capabilities"
+                # todo update to use a constant when the
```
Comment (adriane-cardozo) [inline oneview_redfish_toolkit/api/zone.py]: @data.id should have SPT UUID so we know how to build the Capabilities object, we can update that when adding the other resource blocks

File: oneview_redfish_toolkit/blueprints/util/response_builder.py [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+            mimetype="application/json")
+
+    @staticmethod
+    def success(api_data):
+        return ResponseBuilder.response(api_data, status.HTTP_200_OK)
+
+    @staticmethod
+    def error_by_hp_oneview_exception(exception):
+        error_code = exception.oneview_response['errorCode']
+        http_error_code = HP_ONEVIEW_ERROR_CODE_MAP.get(error_code)\
```
Comment (lhsvobodaj) [inline oneview_redfish_toolkit/blueprints/util/response_builder.py]: The second parameter of a get() is the default value.
```python
HP_ONEVIEW_ERROR_CODE_MAP.get(error_code, status.HTTP_500_INTERNAL_SERVER_ERROR)
```

File: oneview_redfish_toolkit/api/zone.py [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+        self.redfish["@odata.id"] = ZoneCollection.BASE_URI + "/" +\
+            self.redfish["Id"]
+
+        self._validate()
+
+    def fill_resource_blocks(self, profile_template, available_targets):
+        for item in available_targets:
+            self.add_resource_block_item_to_links(item, "serverHardwareUri")
+
+        self.add_resource_block_item_to_links(profile_template, "uri")
```
Comment (lhsvobodaj) [inline oneview_redfish_toolkit/api/zone.py]: Remove the ResourceBlock for the server profile template

File: oneview_redfish_toolkit/tests/api/test_zone.py [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+import json
+
+from oneview_redfish_toolkit.api.zone import Zone
+from oneview_redfish_toolkit.tests.base_test import BaseTest
+
+
+class TestZone(BaseTest):
+    """Tests for Zone class"""
+
+    def setUp(self):
```
Comment (lhsvobodaj) [inline oneview_redfish_toolkit/tests/api/test_zone.py]: You can use `setUpClass` to avoid executing the code below before each test

**Pull request merged**